### PR TITLE
feat: W6 Heartbeat + CLI + E2E Integration

### LINKED ISSUE #6: [Epic] W6: Heartbeat + CLI + E2E Integration
## Overview

Implement the Heartbeat "Notification + Proposal" system, CLI commands, and comprehensive E2E integration testing.

### Heartbeat System (Notification + Proposal Mode)
- node-cron scheduler (default: 30 min interval)
- Context check: in-progress/completed/failed tickets + codebase changes
- Report + proposal generation: summary, new ticket proposals, reverse proposals
- Delivery: Dashboard + Slack/Discord notifications
- User response processing: [Approve All] [Review Individually] [Later]
- Adaptive interval: 15min → 30min → 1hr → 2hrs based on activity level

### CLI Package
- `phalanx init` — Project initialization (create .phalanx/ directory + conventions)
- `phalanx start` — Start daemon
- `phalanx stop` — Stop daemon
- `phalanx status` — Show current status
- `phalanx goal <description>` — Add new goal

### E2E Integration Testing
- Full flow: Goal → decomposition → approval → execution → verification → PR creation
- Heartbeat cycle: scheduled → context check → report → approval
- Reverse proposal: Team Lead suggestion → user approval → ticket creation
- Dashboard real-time updates via WebSocket
- CLI commands end-to-end

### From OpenClaw
- **Heartbeat mechanism**: HEARTBEAT.md checklist, HEARTBEAT_OK silent ACK, timezone-aware windows
- **Lane-based heartbeat**: Separate execution lanes so heartbeats don't block inbound
- **Adaptive scheduling**: Scale interval based on activity level

### From oh-my-opencode
- **Ralph Loop**: Self-referential iteration with completion detection
- **todo-continuation-enforcer**: Auto-resume incomplete tasks
- **Dual-path abort detection**: Event-based + API fallback for robustness

## Success Criteria
- [ ] Heartbeat fires on schedule, generates report
- [ ] Proposals included in heartbeat report
- [ ] User can approve/reject proposals from dashboard
- [ ] Adaptive interval adjusts based on activity
- [ ] All 5 CLI commands work correctly
- [ ] E2E test: goal → PR creation passes
- [ ] E2E test: heartbeat cycle completes

## Dependencies
- #4 W4: Orchestrator + Verification
- #5 W5: Dashboard (for heartbeat delivery)

## Estimated Effort
~5 days (1 developer)

### PR DESCRIPTION
**Pull request by 12OneTwo12**
## Summary

Implements the Heartbeat system, CLI commands, and E2E integration tests — completing the Phase 1 MVP runtime.

Closes #6

### Heartbeat System (`packages/core/src/heartbeat/`)
- **HeartbeatScheduler**: setTimeout + armTimer pattern with drift correction (MAX_TIMER_DELAY=60s), re-entrancy guard, exponential error backoff [30s→60min]
- **ContextChecker**: Collects system state from all repositories (tickets, agents, goals)
- **HeartbeatReporter**: Generates reports + proposals with change detection between cycles
- **AdaptiveIntervalCalculator**: Activity-based strategy — 15min (active) → 30min (normal) → 60min (quiet) → 120min (idle)
- **HeartbeatService**: Facade integrating scheduler + context + reporter with typed events

### CLI (`packages/cli/`)
- `phalanx init` — Create .phalanx/ directory + convention analysis
- `phalanx start` — Start daemon (platform-aware: launchd/systemd/PID file)
- `phalanx stop` — Stop daemon
- `phalanx status` — Display current status
- `phalanx goal <description>` — Add new goal
- **DaemonService abstraction**: LaunchdService (macOS), SystemdService (Linux), PidFileService (fallback)

### E2E Integration Tests
- Goal → decomposition → approval → execution → verification → PR (full pipeline)
- Heartbeat cycle: scheduled → context check → report → adaptive interval
- Reverse proposal: Team Lead suggestion → user approval → ticket creation
- Ticket lifecycle: full state machine transitions

### Dashboard Enhancements
- Heartbeat → EventBus SSE bridge for real-time report delivery
- `PATCH /api/heartbeat/:id` — User response processing (Approve All / Review Individually / Later)
- Enhanced proposals API with reverse proposal support

## Architecture Highlights
- No node-cron dependency — native setTimeout with armTimer pattern for precise control
- Platform-aware daemon management via Strategy pattern (DaemonService interface)
- Exponential error backoff with consecutive error tracking
- Typed EventEmitter events (`HeartbeatServiceEvents`)

## Tests
- **733 tests passing** (86 files) — 600 existing + 133 new
- Heartbeat: scheduler, backoff, adaptive interval, lifecycle, edge cases
- CLI: commands, daemon platform detection
- Integration: 4 E2E test suites
- Dashboard: SSE bridge, heartbeat response API

## Commits (9)
- 6 feature commits + 3 review round improvements

### CODE REVIEW

File: packages/cli/src/commands/goal.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        findByGoalId: () => [],
+      };
+      const ticketStub: Pick<InstanceType<typeof core.TicketRepository>, 'findByEpicId'> = {
+        findByEpicId: () => [],
+      };
+      const manager = new core.GoalManager(
+        goalRepo,
+        epicStub as InstanceType<typeof core.EpicRepository>,
+        ticketStub as InstanceType<typeof core.TicketRepository>,
+      );
```
[the remark this answers is not in the document]
Comment (12OneTwo12) [reply packages/cli/src/commands/goal.ts]: Good catch! Made `epicRepo` and `ticketRepo` optional in the `GoalManager` constructor since they're only needed for `calculateProgress()`. The CLI can now simply call `new GoalManager(goalRepo)` without any stubs. Added a runtime guard in `calculateProgress()` that throws if the repos aren't provided.

File: packages/cli/src/commands/goal.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      const goalRepo = new core.GoalRepository(db.orm);
+      const goals = goalRepo.findAll();
+
+      if (goals.length === 0) {
+        logger.info('No goals found. Use "phalanx goal add" to create one.');
+      } else {
+        logger.heading('Goals');
+        for (const goal of goals) {
+          const statusIcon = goal.status === 'active' ? '●' : goal.status === 'completed' ? '✔' : '◯';
+          console.log(`  ${statusIcon} [${goal.id.slice(0, 8)}] ${goal.description} (${goal.status}, ${goal.progress}%)`);
```
[the remark this answers is not in the document]
Comment (12OneTwo12) [reply packages/cli/src/commands/goal.ts]: Fixed — added a `logger.item()` method and replaced the `console.log` call with it for consistency with the rest of the CLI.

File: packages/cli/src/daemon/launchd-service.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        if (pidMatch) pid = parseInt(pidMatch[1], 10);
+      } catch {
+        // Ignore
+      }
+    }
+
+    return {
+      running,
+      pid,
+      uptime: null,
```
[the remark this answers is not in the document]
Comment (12OneTwo12) [reply packages/cli/src/daemon/launchd-service.ts]: Good point! Implemented uptime calculation using `ps -p <PID> -o etime=` with a `parseEtime()` helper that converts the elapsed time format (e.g. `2-03:45:12`) into a human-readable string (e.g. `2d 3h 45m`), matching the SystemdService behavior.

File: packages/dashboard/src/app/api/heartbeat/[id]/route.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+type HeartbeatAction = 'acknowledge' | 'approve_all' | 'review' | 'later';
+
+const ACTION_TO_STATUS: Record<HeartbeatAction, 'acknowledged' | 'acted'> = {
+  acknowledge: 'acknowledged',
+  approve_all: 'acted',
+  review: 'acted',
+  later: 'pending', // keeps status as pending
+};
+
+const VALID_ACTIONS: HeartbeatAction[] = ['acknowledge', 'approve_all', 'review', 'later'];
```
[the remark this answers is not in the document]
Comment (12OneTwo12) [reply packages/dashboard/src/app/api/heartbeat/[id]/route.ts]: Agreed — `review` means the user wants to navigate to individual review, not that they've completed the action. Changed it to keep `pending` status while still emitting the `heartbeat:response` event so the UI can handle the navigation.
